File: leetcode/0451. Sort Characters By Frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        return "".join(
            [
                char * freq
                for char, freq in sorted(
                    Counter(s).items(), key=lambda e: e[1], reverse=True
                )
            ]
        )

    def frequencySort2(self, s: str) -> str:
        count = dict()
        for char in s:
            count[char] = count.get(char, 0) + 1

        ans = ""
        for char, freq in sorted(count.items(), key=lambda e: e[1], reverse=True):
            ans += char * freq
        return ans

    def frequencySort3(self, s: str) -> str:
        return "".join([char * freq for char, freq in Counter(s).most_common()])

File: leetcode/0451. Sort Characters By Frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> count;
        for (const char& c : s) {
            count[c] += 1;
        }

        vector<pair<int, char>> countsAndChars;
        for (const auto& [ch, freq] : count) {
            countsAndChars.push_back(make_pair(freq, ch));
        }

        sort(countsAndChars.begin(), countsAndChars.end(),
             greater<pair<int, char>>());

        string ans;
        for (const auto& [freq, ch] : countsAndChars) {
            ans += string(freq, ch);
        }
        return ans;
    }
};